syntax = "proto3";

package rk.projects.counter_app;

message CounterLogEntry{
  enum CommandType{
    INCR = 0;
    DECR = 1;
  }

  string key = 1;
  int64 val = 2;
  CommandType commandType = 3;
  string description = 4;
}

message CounterLogEntries {
  repeated CounterLogEntry entries = 1;
}

message CounterAppSnapshot{
  int64 log_id = 1;
  map<string, int64> values = 2;
}